yt_idefix/data_structures.py

```diff
@@ -232,14 +252,58 @@ def __init__(
             default_species_fields=default_species_fields,
         )
 
-        self.inifile = inifile
-        self._parse_inifile()
-
         self.storage_filename = None
 
         # idefix does not support grid refinement
         self.refine_by = 1
 
+    def _get_inifile(self, inifile: str | os.PathLike[str] | None) -> str:
+        if inifile is not None:
+            if os.path.isabs(inifile):
+                return os.fspath(inifile)
+            else:
+                return os.path.abspath(os.path.join(self.directory, inifile))
+
+        root_dir = Path(self.directory)
+        if (
+            len(candidates := list(root_dir.glob("*.ini"))) == 1
+            and (file := candidates[0]).name == self._default_inifile
+        ):
+            return str(file.absolute())
+        else:
+            # TODO: writeme
+            ytLogger.warning("...")
+            return ""
+
+    def _get_definitions_header(
+        self, definitions_header: str | os.PathLike[str] | None
+    ) -> str:
+        # TODO: be careful with name collisions (we already have a '_get_header' method)
+        if definitions_header is not None:
+            if os.path.isabs(definitions_header):
+                return os.fspath(definitions_header)
+            else:
+                return os.path.abspath(os.path.join(self.directory, definitions_header))
+
+        ext: str
+        if self._header_type is Header.C:
+            ext = "h"
+        elif self._header_type is Header.CPP:
+            ext = "hpp"
+        else:
+            assert_never(self._header_type)
+
+        root_dir = Path(self.directory)
+        if (
+            len(candidates := list(root_dir.glob(f"*.{ext}"))) == 1
+            and (file := candidates[0]).name == f"definitions.{ext}"
+        ):
```

Comment:
As I know, the header file `definitions.h` cannot be replaced with another file by users in Pluto unless modifying the source code, don't know about the idefix. Is it enough to just check the existence of `definitions.h` or `definitions.hpp` here?

Reply:
Ah, thanks for pointing this out. Idefix indeed allows this file to be named arbitrarily, so I assumed Pluto did too. Do you think it would impact the user experience if I keep it that way still, so as to reduce code duplication ?

Reply:
I think it's fine!